physical/consul_test.go

```diff
@@ -3,16 +3,424 @@ package physical
+			t.Fatalf("Expected config %s to not fail: %v", test.name, err)
+		}
+
+		c, ok := be.(*ConsulBackend)
+		if !ok {
+			t.Fatalf("Expected ConsulBackend: %s", test.name)
+		}
+		c.disableRegistration = true
+
+		var shutdownCh ShutdownChannel
+		if err := c.RunServiceDiscovery(shutdownCh, test.advertiseAddr); err != nil {
+			t.Fatalf("bad: %v", err)
+		}
+
+		if test.checkTimeout != c.checkTimeout {
+			t.Errorf("bad: %v != %v", test.checkTimeout, c.checkTimeout)
+		}
+
+		if test.path != c.path {
+			t.Errorf("bad: %s %v != %v", test.name, test.path, c.path)
+		}
+
+		if test.service != c.serviceName {
+			t.Errorf("bad: %v != %v", test.service, c.serviceName)
+		}
+
+		if test.address != c.consulClientConf.Address {
+			t.Errorf("bad: %s %v != %v", test.name, test.address, c.consulClientConf.Address)
+		}
+
+		if test.scheme != c.consulClientConf.Scheme {
+			t.Errorf("bad: %v != %v", test.scheme, c.consulClientConf.Scheme)
+		}
+
+		if test.token != c.consulClientConf.Token {
+			t.Errorf("bad: %v != %v", test.token, c.consulClientConf.Token)
+		}
+
+		// FIXME(sean@): Unable to test max_parallel
+		// if test.max_parallel != cap(c.permitPool) {
+		// 	t.Errorf("bad: %v != %v", test.max_parallel, cap(c.permitPool))
+		// }
+	}
+}
+
+func TestConsul_serviceTags(t *testing.T) {
+	tests := []struct {
+		active bool
+		tags   []string
+	}{
+		{
+			active: true,
+			tags:   []string{"active"},
+		},
+		{
+			active: false,
+			tags:   []string{"standby"},
+		},
+	}
+
+	for _, test := range tests {
+		tags := serviceTags(test.active)
+		if !reflect.DeepEqual(tags[:], test.tags[:]) {
+			t.Errorf("Bad %v: %v %v", test.active, tags, test.tags)
+		}
+	}
+}
+
+func TestConsul_parseAdvertiseAddr(t *testing.T) {
+	tests := []struct {
+		addr string
+		host string
+		port int
+		pass bool
+	}{
+		{
+			addr: "http://127.0.0.1:8200/",
+			host: "127.0.0.1",
+			port: 8200,
+			pass: true,
+		},
+		{
+			addr: "http://127.0.0.1:8200",
+			host: "127.0.0.1",
+			port: 8200,
+			pass: true,
+		},
+		{
+			addr: "https://127.0.0.1:8200",
+			host: "127.0.0.1",
+			port: 8200,
+			pass: true,
+		},
+		{
+			addr: "unix:///tmp/.vault.addr.sock",
+			host: "/tmp/.vault.addr.sock",
+			port: 0,
+			pass: true,
+		},
+		{
+			addr: "127.0.0.1:8200",
+			pass: false,
+		},
+		{
+			addr: "127.0.0.1",
+			pass: false,
+		},
+	}
+	for _, test := range tests {
+		c := testConsulBackend(t)
+		host, port, err := c.parseAdvertiseAddr(test.addr)
+		if test.pass {
+			if err != nil {
+				t.Fatalf("bad: %v", err)
+			}
+		} else {
+			if err == nil {
+				t.Fatalf("bad, expected fail")
+			} else {
+				continue
+			}
+		}
+
+		if host != test.host {
+			t.Fatalf("bad: %v != %v", host, test.host)
+		}
+
+		if port != test.port {
+			t.Fatalf("bad: %v != %v", port, test.port)
+		}
+	}
+}
+
+func TestConsul_AdvertiseActive(t *testing.T) {
+	addr := os.Getenv("CONSUL_HTTP_ADDR")
+	if addr == "" {
+		t.Skipf("No consul process running, skipping test")
```

Comment:
NM. 
"A test ends when its Test function returns or calls any of the methods FailNow, Fatal, Fatalf, SkipNow, Skip, or Skipf"